test/compilable/commontype.d

```diff
@@ -0,0 +1,491 @@
+
+static assert(Error!( C, void* ));
+static assert(Error!( void*, C ));
+
+static assert(is( X!( C, C ) == C ));
+static assert(is( X!( C, B ) == B ));
+static assert(is( X!( C, I ) == I ));
+static assert(is( X!( C, D ) == B ));
+static assert(is( X!( C, K ) == Object ));
+
+static assert(is( X!( C, SC ) == C ));
+static assert(is( X!( C, SB ) == B ));
+static assert(is( X!( C, SI ) == I ));
+static assert(is( X!( C, SD ) == B ));
+static assert(is( X!( C, SK ) == Object ));
+
+static assert(is( X!( C, immutable(C) ) == const(C) ));
+static assert(is( X!( C, immutable(I) ) == const(I) ));
+static assert(is( X!( C, immutable(B) ) == const(B) ));
+static assert(is( X!( C, immutable(D) ) == const(B) ));
+static assert(is( X!( C, immutable(K) ) == const(Object) ));
+
+static assert(Error!( C, immutable(SC) )); // should work
+static assert(Error!( C, immutable(SI) )); // should work
+static assert(Error!( immutable(SI), C )); // should work
+static assert(Error!( C, immutable(SB) )); // should work
+static assert(Error!( C, immutable(SD) )); // should work
+static assert(Error!( C, immutable(SK) )); // should work
+
+static assert(is( X!( const(C), C ) == const(C) ));
+static assert(is( X!( const(C), I ) == const(I) ));
+static assert(is( X!( const(C), B ) == const(B) ));
+static assert(is( X!( const(C), D ) == const(B) ));
+static assert(is( X!( const(C), K ) == const(Object) ));
+
+static assert(is( X!( const(C), SC ) == const(C)));
+static assert(Error!( const(C), SI )); // should work
+static assert(is( X!( const(SI), const(C) ) == const(I) )); // should work
+static assert(is( X!( const(C), SB ) == const(B)));
+static assert(is( X!( const(C), SD ) == const(B)));
+static assert(is( X!( const(C), SK ) == Object)); // `const`
+
+static assert(is( X!( SiC, SC ) == const(C) ));
+
+/******************************
+ * Structs
+ */
+
+static assert(is( X!( S, S ) == S ));
+static assert(is( X!( S, immutable(S) ) == const(S) ));
+static assert(Error!( S, shared(S) ));
+static assert(is( X!( Si, Si ) == Si ));
+static assert(is( X!( Si, int ) == int ));
+static assert(is( X!( int, Si ) == int ));
+static assert(is( X!( Si, Si2 ) == int ));
+
+static assert(is( X!( int, Sl ) == long ));
+static assert(is( X!( Si, Sl ) == long ));
+
+
+/******************************
+ * Vectors
+ */
+
+version(D_SIMD)
+{
+
+static assert(is( X!( int4, int4 ) == int4));
+static assert(is( X!( int4, const(int4) ) == const(int4)));
+static assert(is( X!( int4, immutable(int4) ) == const(int4)));
+static assert(is( X!( __vector(const(int)[4]), int4 ) == int4));
+static assert(is( X!( int4, __vector(const(int)[4]) ) == int4));
+
+static assert(Error!( byte16, void16 ));
+static assert(Error!( int4, void16 ));
+static assert(Error!( float4, void16 ));
+static assert(Error!( byte16, ubyte16 ));
+static assert(Error!( short8, ushort8 ));
+static assert(Error!( int4, uint4 ));
+static assert(Error!( int4, float4 ));
+version (D_AVX)
+{
+static assert(Error!( long4, ulong4 ));
```

Comment:
What about the types here https://github.com/dlang/druntime/blob/715ccf41b19d64a89e434798acf585005fdd6168/src/core/simd.d#L55-L65?

Reply:
I didn't put all of them here, since they're all supposed to fail.